week2/homework/src/index.js

```diff
@@ -1,3 +1,76 @@
 'use strict';
 
 // TODO: Write the homework code in this file
+
+const { readFile, writeFile } = require('fs');
+const { promisify } = require('util');
+
+const readFileWithPromise = promisify(readFile);
+const writeFileWithPromise = promisify(writeFile);
+
+const args = process.argv.slice(2);
+const cmd = args[0];
+
+const TODO_PATH = 'todo.json';
+
+function readTodos() {
+  return readFileWithPromise(TODO_PATH, 'utf8')
+    .then(JSON.parse)
+    .catch(() => ([]));
+}
+
+function writeTodos(todos) {
+  return writeFileWithPromise(TODO_PATH, JSON.stringify(todos, null, 2));
+}
+
+async function main() {
+  switch (cmd) {
+    case 'add':
+      const text = args[1];
+      const todos = await readTodos();
+      todos.push({
+        text,
+        done: false
+      });
+      await writeTodos(todos);
+      const newTodos = await readTodos();
+      console.log(newTodos);
+      break;
+
+    case 'remove':
+      const remove = await readTodos();
+      remove.splice(args[1] - 1, 1);
```

Comment:
What if I call `remove` with invalid index like `-1000`, `cat` or even `0`? You're not handling those cases.